lib/puppet/provider/package/gem.rb

```diff
@@ -112,35 +132,35 @@ def install(useversion = true)
       case uri.scheme
       when nil
         # no URI scheme => interpret the source as a local file
-        command << source
+        command_options << source
       when /file/i
-        command << uri.path
+        command_options << uri.path
       when 'puppet'
         # we don't support puppet:// URLs (yet)
         raise Puppet::Error.new(_("puppet:// URLs are not supported as gem sources"))
       else
         # check whether it's an absolute file path to help Windows out
         if Puppet::Util.absolute_path?(source)
-          command << source
+          command_options << source
         else
           # interpret it as a gem repository
-          command << "--source" << "#{source}" << resource[:name]
+          command_options << "--source" << "#{source}" << resource[:name]
         end
       end
     else
-      command << "--no-rdoc" << "--no-ri" << resource[:name]
+      command_options << resource[:name]
```

Comment:
Previously we only appended `--no-rdoc` & `--no-ri` when `resource[:source]` was `nil`. Is it safe to always append them as is done in this PR? I tested a simple case that seems ok, but what about the others?

```
$ gem install ./colorize-0.8.1.gem --no-document
Successfully installed colorize-0.8.1
1 gem installed
```

Reply:
The issue you pointed out, and Charlie confirmed, was that we only want to add `resource[:name]` (the name of the gem package) when `:source` is specified and we interpret `:source` as a gem repository. 

I did some research and found nothing about the `--no-` documentation flags being incompatible with the other cases, and here are tests:

```
# gem repository
[root@pe-master ~]# gem install --no-document --source https://rubygems.org colorize 
Successfully installed colorize-0.8.1
1 gem installed

# absolute file path
[root@pe--master ~]# gem install --no-document /root/colorize-0.8.1.gem 
Successfully installed colorize-0.8.1
1 gem installed
```